pkgs/development/tools/ilspycmd/default.nix

```
@@ -1,52 +1,59 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, buildDotnetModule
-, dotnetCorePackages
-, powershell
-, autoSignDarwinBinariesHook
-, glibcLocales
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  buildDotnetModule,
+  dotnetCorePackages,
+  powershell,
+  autoSignDarwinBinariesHook,
+  glibcLocales,
 }:
-buildDotnetModule rec {
+buildDotnetModule (finalAttrs: rec {
   pname = "ilspycmd";
-  version = "8.0";
+  version = "9.0-preview3";
 
   src = fetchFromGitHub {
     owner = "icsharpcode";
     repo = "ILSpy";
     rev = "v${version}";
-    hash = "sha256-ERBYXgpBRXISfqBSBEydEQuD/5T1dvJ+wNg2U5pKip4=";
+    hash = "sha256-7cPXFaEKr76GtqcNsKx7tstRUeTpSTF8ggxbyEnQa9M=";
   };
 
-  nativeBuildInputs = [
-    powershell
-  ] ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [ autoSignDarwinBinariesHook ];
+  nativeBuildInputs =
+    [
+      powershell
+    ]
+    ++ lib.optionals (stdenvNoCC.isDarwin && stdenvNoCC.isAarch64) [
+      autoSignDarwinBinariesHook
+    ];
 
   # https://github.com/NixOS/nixpkgs/issues/38991
   # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
-  env.LOCALE_ARCHIVE = lib.optionalString stdenv.hostPlatform.isLinux "${glibcLocales}/lib/locale/locale-archive";
+  env.LOCALE_ARCHIVE = lib.optionalString stdenvNoCC.isLinux "${glibcLocales}/lib/locale/locale-archive";
```

My bad for suggesting the opposite before
See https://github.com/NixOS/nixpkgs/pull/341407
```suggestion
  env.LOCALE_ARCHIVE = lib.optionalString stdenvNoCC.hostPlatform.isLinux "${glibcLocales}/lib/locale/locale-archive";
```
(and all the other instances of platform checking without `hostPlatform`)

---

Done.